skopt/space/space.py

```diff
@@ -550,6 +552,49 @@ def is_real(self):
         """
         return all([isinstance(dim, Real) for dim in self.dimensions])
 
+    @classmethod
+    def from_yaml(cls, yml_path, namespace=None):
+        """Create Space from yaml configuration file
+
+        Parameters
+        ----------
+        * `yml_path` [str]:
+           Full path to yaml configuration file
+        * `namespace` [str, default=None]:
+           Namespace within configuration file to use
+
+        Returns
+        -------
+        * `space` [Space]:
+           Instantiated Space object
+        """
+        with open(yml_path, 'rb') as f:
+            cfg = yaml.load(f)
+
+        this_module = sys.modules[__name__]
+        # Dynamically create list of subclass names for Dimension
+        dimension_classes = ['Real', 'Integer', 'Categorical']
+
+        # Extract space options for configuration file
+        options_name = namespace or list(cfg.keys())[0]
```

Comment:
`list(cfg.keys())[0]` could be `next(config)`? Not sure which is nicer.

What do you think about making this into an explicit:
```python
if namespace is None:
  namespace = next(config) # or list(config.keys())[0]
```
explicitly comparing to `None` and overwriting `namespace`. Feels like a "standard pattern" for dealing with default arguments

Reply:
i believe list(cfg.keys())[0] is equivalent to next(iter(cfg.keys())), hard to say if there's an improvement, maybe it's slightly nicer not to deal with indexing /shrug

agreed on the namespace, i will clean this section up a bit